### DISCUSSION TITLE
根据docker-compose安装的单机版milvus 2.3.2，使用官方文档访问insert接口404

### DISCUSSION THREAD

**Discussion by dushulin** [milvus-io/milvus#28526] [Q&A and General discussion] [answered]
参考这个文档https://milvus.io/docs/v2.2.x/insert_data.md，发现curl访问部署的单机版milvus 404。是否有组件没有部署，还是官方文档的api有误？
部署了minio、etcd、milvus-standalone

**Comment by yhmo** [comment, marked as answer]
1. `docker ps -a` to show all containers, `docker stop [container_id]` and `docker rm [container_id]` to close all the containers

2. download the yaml
`wget https://github.com/milvus-io/milvus/releases/download/v2.3.3/milvus-standalone-docker-compose.yml -O docker-compose.yml`

3. `docker-compose up -d`

4. create a collection
```
curl --request POST \
     --url "localhost:19530/v1/vector/collections/create" \
     --header "Authorization: Bearer}" \
     --header "accept: application/json" \
     --header "content-type: application/json" \
     -d '{
       "dbName": "default",   
       "collectionName": "medium_articles",
       "dimension": 4,
       "metricType": "L2",
       "primaryField": "id",
       "vectorField": "vector"
      }'

```

5. insert an entity
```
curl --request POST \
     --url "localhost:19530/v1/vector/insert?db_name=default" \
     --header "Authorization: Bearer" \
     --header "accept: application/json" \
     --header "content-type: application/json" \
     -d '{
          "collectionName": "medium_articles", 
          "data": [{ 
               "vector": [0.23254494, 0.01374953, 0.88497432, 0.05292784],
               "title": "Top 10 2020",
               "link": "4462eb7d8d3e"
          }]
     }'
```

**Reply by dushulin** [reply]
官方文档的2.3.x的api是：
# insert an entity to a collection
curl -X 'POST' \
  '${MILVUS_HOST}:${MILVUS_PORT}/v1/vector/insert'  \
  -H 'Authorization: Bearer ${TOKEN}' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
         "collectionName": "collection1",
         "data": {
             "vector": [0.1, 0.2, 0.3],
             "name": "tom",
             "email": "tom@zilliz.com",
             "date": "2023-04-13"
          }
     }'
跟你上面的insert an entity的接口不一致，是不是需要修改一下官方文档给出的api？

**Reply by yhmo** [reply]
我只是加了个“?db_name=default”，可以去掉的。

api是相同的，只不过输入的数据不一样，官方的建表示例是定义了256维的向量，如果在这insert里面写一个256维向量太浪费页面。所以这示例里是假设你知道向量维度这个概念的，定义了多少维度的表，就应该插入多少维度的向量。示例里只写了3个维度示意示意即可。

我的示例里建的表的向量是4维度的，所以这样insert
```
curl --request POST \
     --url "localhost:19530/v1/vector/insert" \
     --header "Authorization: Bearer" \
     --header "accept: application/json" \
     --header "content-type: application/json" \
     -d '{
          "collectionName": "medium_articles", 
          "data": [{ 
               "vector": [0.1, 0.2, 0.3, 0.4],
               "name": "tom",
               "email": "tom@zilliz.com",
               "date": "2023-04-13"
          }]
     }'
```

**Reply by dushulin** [reply]
搭建了集群版后，api是正常的了 感谢
